core/core/util.go

```diff
@@ -0,0 +1,83 @@
+package xchaincore
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/xuperchain/xuperchain/core/pb"
+)
+
+func produceBlockEvent(msgChan chan *pb.Event, block *pb.InternalBlock, bcname string) {
+	blockStatus := &pb.BlockStatusInfo{
+		Bcname:   bcname,
+		Proposer: fmt.Sprintf("%s", block.GetProposer()),
+		Height:   block.GetHeight(),
+	}
+	blockEvent := &pb.BlockEvent{
+		Block: block,
+	}
+	payload, _ := proto.Marshal(blockEvent)
```

Comment:
payload can be nil ?

Reply:
payload being nil can pass but not expected.